blocklylib-core/src/main/java/com/google/blockly/model/Block.java

```diff
@@ -862,12 +867,32 @@ public Connection getParentConnection() {
      * @return The block connected to the output or previous {@link Connection}, if present.
      *         Otherwise null.
      */
+    @Nullable
     public Block getParentBlock() {
         Connection parentConnection = getParentConnection();
         return parentConnection == null ? null : parentConnection.getBlock();
     }
 
     /**
+     * Sets a event callback that will follow the block to any workspace (or workspace equivalent:
```

Comment:
'follow the block to any workspace' implies that this is registered per workspace, but in code it's registered for that blockly instance (on the Controller). Consider rewording this to something like,
"Sets an event callback that will receive {@link BlocklyEvent}s for the lifetime of the block."